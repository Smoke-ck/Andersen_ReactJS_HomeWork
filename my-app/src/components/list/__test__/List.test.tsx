import ConnectedList, { List } from "../List"
import { shallow, render, mount } from "enzyme";
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallowToJson } from 'enzyme-to-json';
import thunk from "redux-thunk";

 const props: any = {
    todos: [{
      title: "hi",
      completed: false,
      favorite: false,
      id: 50,
    }, {
      title: "hi",
      completed: false,
      favorite: false,
      id: 51,
    }],
    onItemToggle: jest.fn(),
    onItemFavorite: jest.fn(),
    onItemDelete: jest.fn(),
    onUpdate: jest.fn()
  };
  
  describe('List',()=>{
    
    it('should show container div', () => {
      const render =  shallow(<List {...props}/>)
      expect(render.find('[className="list"]').length).toEqual(1)
    });

});

const todos = [{
  title: "hi",
  completed: false,
  favorite: false,
  id: 50,
}, {
  title: "hi",
  completed: true,
  favorite: true,
  id: 51,
}, {
  title: "hi",
  completed: false,
  favorite: false,
  id: 52,
}, {
  title: "hi",
  completed: true,
  favorite: false,
  id: 53,
}]


const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

const store = mockStore({
  getState: jest.fn(() => ({
    todos:todos,
    todosFilter:{value:'all'}
  })),
  dispatch: jest.fn()
})
describe('ConnectedList container', () => {
  let wrapper:any;
  it('render wrapped into provider with mocked store', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <ConnectedList {...todos}/>
      </Provider>
    )
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  })
})