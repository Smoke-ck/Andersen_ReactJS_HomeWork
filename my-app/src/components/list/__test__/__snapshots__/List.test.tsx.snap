// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedList container render wrapped into provider with mocked store 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Connect(List)
    0={
      Object {
        "completed": false,
        "favorite": false,
        "id": 50,
        "title": "hi",
      }
    }
    1={
      Object {
        "completed": true,
        "favorite": true,
        "id": 51,
        "title": "hi",
      }
    }
    2={
      Object {
        "completed": false,
        "favorite": false,
        "id": 52,
        "title": "hi",
      }
    }
    3={
      Object {
        "completed": true,
        "favorite": false,
        "id": 53,
        "title": "hi",
      }
    }
  />
</ContextProvider>
`;
